src/xSTIR/cSTIR/stir_data_containers.h

```diff
@@ -469,30 +470,91 @@ class PETImageData : public aDataContainer<float> {
 	PETImageData(const PETImageData& image)
 	{
 		_data.reset(image.data().clone());
+		_patient_coord_geometrical_info.reset(
+			new GeometricalInfo(image.get_patient_coord_geometrical_info()));
 	}
 	PETImageData(const PETAcquisitionData& ad)
 	{
 		_data.reset(new Voxels3DF(*ad.get_proj_data_info_sptr()));
+		// TODO: How can we get the geometrical info?
 	}
+	// Perhaps this can no longer exist, as we can't know the size and spacing of a DiscretisedDensity
```

Comment:
could check type of arg (`dynamic_cast<const Voxels3DF *>(&image)!=0`). if it doesn't match, throw exception

Reply:
Interesting. Is there a way to tell the compiler to always use the overloaded call with the lowest hierarchy of inheritance for an object?

Reply:
you can get the dimensions and voxel sizes out of an Image3DF - see `PETImageData::get_dimensions` and `PETImageData::get_voxel_sizes` in `stir_data_containers.cpp`

Reply:
What do voxel_sizes mean for a non-cartesian-voxelised image? I have reimplemented this anyway